src/pystatis/db.py

```diff
@@ -1,61 +1,66 @@
 """Module provides functions to set the active database and get active database properties."""
 import logging
+import re
 
 from pystatis import config
+from pystatis.cache import normalize_name
 from pystatis.exception import PystatisConfigError
 
 logger = logging.getLogger(__name__)
 
-
-def set_db(name: str) -> None:
-    """Set the active database.
-
+def match_db(name: str) -> str:
+    """Match item code to database.
+    
     Args:
-        name (str): Name of the database. Must be one of the supported databases.
-        See `pystatis.config.get_supported_db()`.
-    """
-    if name.lower() not in config.get_supported_db():
-        raise ValueError(
-            f"Database {name} not supported! Please choose one of {', '.join(config.get_supported_db())}"
-        )
-    config.config.set("settings", "active_db", name.lower())
-
-    if not get_db_user() or not get_db_pw():
-        logger.critical(
-            "No credentials for %s found. Please run `setup_credentials()`.",
-            name,
-        )
+        name (str): Query parameter 'name' corresponding to item code.
 
+    Returns: 
+        db_name (str): Name of matching database. 
+    """
+    supported_dbs = config.get_supported_db()
+    regex_db = config.get_regex()
 
-def get_db() -> str:
-    """Get the active database."""
-    active_db = config.config.get("settings", "active_db")
+    # Strip optional leading * and trailing job id
+    name = normalize_name(name).lstrip("*")
 
-    if not active_db:
-        raise PystatisConfigError(
-            "No active database set! Please run `set_db()`."
-        )
+    # Get list of matching dbs
+    db_match = [idb for idb, irx in zip(supported_dbs, regex_db) if re.match(irx, name)]
 
-    return active_db
+    if db_match:
+        # If more than one db matches it must be a Cube (provided all regexing works as intended).
+        # --> Choose db based on available credentials.
+        if len(db_match) > 1:
+            for check_db in db_match:
+                if get_db_user(check_db) and get_db_pw(check_db):
+                    return check_db
+            else:
+                raise PystatisConfigError(
+                    "You lack the necessary credentials to access this item. " \
+                    "Please run setup_credentials()."
+                )
+        else:
+            return db_match[0]
```

Comment:
just found a small error: When there is only a single match, we return it without checking the credentials. But I believe it would be better to move the check out of this function, anyway. The function should handle matching the db without checking for the credentials, I guess this is a better job for the calling function or another helper method